lib/src/common.ts

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useState } from "react";
+
+import type {
+  SbGatsbyStory,
+  StoryblokBridgeConfigV2
+} from '../types'
+
+
+export const useStoryblokBridge = (
+  id: Number,
+  cb: (newStory) => void,
+  options: StoryblokBridgeConfigV2 = {},
+) => {
+  const isServer = typeof window === "undefined";
+  const isBridgeLoaded =
+    !isServer && typeof window.storyblokRegisterEvent !== "undefined";
+  const storyId = new URL(window.location?.href).searchParams.get(
+    "_storyblok"
+  );
+  const inStory = +storyId === id;
+
+  if (!isBridgeLoaded || !inStory) {
+    return;
+  }
+
+  if (!id) {
+    console.warn("Story ID is not defined. Please provide a valid ID.");
+    return;
+  }
+
+  const refreshDevContent = () => {
+    return fetch('__refresh', { method: 'post' });
+  }
+
+  window.storyblokRegisterEvent(() => {
+    const sbBridge = new window.StoryblokBridge(options);
+    sbBridge.on(["input", "published", "change"], async (event) => {
+      if (event.action === "input" && event.story.id === id) {
+        cb(event.story);
+      } else if (
+        (event.action === "change" || event.action === "published") &&
+        (event.storyId as number) === id
+      ) {
+        await refreshDevContent();
```

Comment:
@schabibi1 have you checked that this is working as expected in production? (As expected means: this shouldn't be called, as it's expected to be run on development)

I'm assuming that, as we have the `isBridgeLoaded` variable above, `refreshDevContent` will run only when the bridge is active.

However, it can happen that they keep it enabled in production, so this will be run (when accesed from the visual editor). Can we check that nothing breaks in that situation?

Reply:
Tested and all works as expected @alexjoverm